packages/realm-react/src/__tests__/RealmProvider.test.tsx

```diff
@@ -43,317 +48,556 @@ const catSchema: Realm.ObjectSchema = {
+      const toggleRefPath = getByTestId("toggleRefPath");
+
+      // Wait a tick for the RealmProvider to set the reference and then call a function that uses the ref
+      await act(async () => {
+        await new Promise<void>((resolve) => setTimeout(resolve, 0));
+        fireEvent.press(toggleRefPath);
+      });
+
+      const realmRefPathText = await waitFor(() => queryByTestId("realmRefPath"));
+
+      expect(realmRefPathText).toHaveTextContent("testPath.realm", { exact: false });
     });
-  });
-  it("handle state changes to its configuration", async () => {
-    const RealmComponent = () => {
-      const realm = useRealm();
-      return <Text testID="schemaName">{realm.schema[0].name}</Text>;
-    };
-    const App = () => {
-      const [schema, setSchema] = useState(dogSchema);
-      return (
-        <>
-          <View testID="firstRealmProvider">
-            <RealmProvider schema={[schema]}>
-              <RealmComponent />
+
+    // TODO: Now that local realm is immediately set, the fallback never renders.
+    // We need to test synced realm in order to produce the fallback
+    describe.skip("initially renders a fallback, until realm exists", () => {
+      it("as a component", async () => {
+        const App = () => {
+          return (
+            <RealmProvider fallback={() => <View testID="fallbackContainer" />}>
+              <View testID="testContainer" />
             </RealmProvider>
-          </View>
-          <Button testID="changeSchema" title="change schema" onPress={() => setSchema(catSchema)} />
-        </>
-      );
-    };
-    const { getByTestId } = render(<App />);
-    const schemaNameContainer = await waitFor(() => getByTestId("schemaName"));
-    const changeSchemaButton = getByTestId("changeSchema");
+          );
+        };
+        const { queryByTestId } = render(<App />);
+
+        expect(queryByTestId("fallbackContainer")).not.toBeNull();
+        expect(queryByTestId("testContainer")).toBeNull();
+
+        await waitFor(() => queryByTestId("testContainer"));
+
+        expect(queryByTestId("fallbackContainer")).toBeNull();
+        expect(queryByTestId("testContainer")).not.toBeNull();
+      });
+
+      it("as an element", async () => {
+        const Fallback = <View testID="fallbackContainer" />;
+        const App = () => {
+          return (
+            <RealmProvider fallback={Fallback}>
+              <View testID="testContainer" />
+            </RealmProvider>
+          );
+        };
+        const { queryByTestId } = render(<App />);
+
+        expect(queryByTestId("fallbackContainer")).not.toBeNull();
+        expect(queryByTestId("testContainer")).toBeNull();
+
+        await waitFor(() => queryByTestId("testContainer"));
+
+        expect(queryByTestId("fallbackContainer")).toBeNull();
+        expect(queryByTestId("testContainer")).not.toBeNull();
+      });
+    });
+  });
+
+  describe("with an existing Realm instance", () => {
```

Comment:
tried to cover a good ground with testing but let me know if there's anything that may be worthwhile to test more.